applications/FluidDynamicsApplication/custom_elements/fluid_element.cpp

```diff
@@ -617,6 +617,78 @@ void FluidElement<TElementData>::CalculateGeometryData(Vector &rGaussWeights,
         rGaussWeights[g] = DetJ[g] * IntegrationPoints[g].Weight();
 }
 
+template <class TElementData>
+void FluidElement<TElementData>::Calculate(
+    const Variable<double> &rVariable,
+    double &rOutput,
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    Element::Calculate(rVariable, rOutput, rCurrentProcessInfo);
```

Comment:
```suggestion
    KRATOS_ERROR << "Asking for the elemental calculation of "  << rVariable.Name() << ". This is not implemented." << std::endl;
```

As the base class `Element` has an empty implementation of all the `Calculate` I'd throw a `KRATOS_ERROR` if we ask for a non-implemented variable calculation. This avoids lots of debugging headaches.

Reply:
Would it be ok if this error is raised in the Element?

Reply:
I guess it's not ok. Probably for this PR, it's better to restrict the changes to FluidDynamicsApp.

Reply:
I don't think it is possible to change this in the baseclass

It can be discussed (forever :smirk:) whether throwing an error or not is a bug or a feature, but unless there is a very good reason I would say that the behavior should be consistent with the core (@rubenzorrilla)

I guess the reason this cannot throw is that if you have different types of elements
one element implements this variable, one doesn't
=> when requesting output, throwing here would crash it even though it is a valid usecase

Reply:
Interesting....

Anyways, raising the error in the `Element` resulted in test failures. I have reverted the change and the behavior of  `Calculate` in case of non-implementation in `FluidElement` mimics the behavior in `Element`. I think this should be good enough for now.

Reply:
> I don't think it is possible to change this in the baseclass
> 
> It can be discussed (forever smirk) whether throwing an error or not is a bug or a feature, but unless there is a very good reason I would say that the behavior should be consistent with the core (@rubenzorrilla)
> 
> I guess the reason this cannot throw is that if you have different types of elements
> one element implements this variable, one doesn't
> => when requesting output, throwing here would crash it even though it is a valid usecase

Precisely. However this is not the case in the fluid, which always uses a single type of element. As this is intentionally done in that way, I think throwing the error may have sense in this case.

Reply:
> Precisely. However this is not the case in the fluid, which always uses a single type of element. As this is intentionally done in that way, I think throwing the error may have sense in this case.

Sorry but I am little confused here, because `QSVMS`, `DVMS`, `FIC` also don't throw error. Should that also be addressed in this PR, or leave it as it is?

Reply:
OK. Let's leave it as it is to be consistent to what we have right now. If at some point we change our minds, we can always implement the error thrown in the base fluid element.